### ISSUE TITLE
closed property is not set when "Error: Client closed" error occurs

### ISSUE THREAD

**Issue by jeremyml** [patrickjuchli/basic-ftp#105]
If the network connection to the FTP server stops working, then basic-ftp throws an error with the message `Client is closed`.  However, the `closed` property of the client remains false.

```
const basicFtp = require("basic-ftp")
var client = new basicFtp.Client();
var ftpAccess = await lib1.startFtpAccess(client);
// remove network connection to the FTP server
try {
  await client.features(); 
} catch(err) {
  console.log(client.closed); // always true, despite error message
  console.error(err); // Error: Client is closed
}
```
This is not expected behavior.  I would expect the `closed` property status to change when the error message says `Client id closed`.

What is the best way to check for a closed connection?

**Comment by patrickjuchli** [comment]
Interesting, I have a couple of questions:

- What version of Node are you using? 
- What do you mean by "remove network connection to the FTP server", what happened exactly?
- What value does `client.ftp.socket.remoteAddress` contain when you expect the client to be closed?
- I'm not sure I understand the comment "always true, despite error message" – client.closed is always true in the corresponding line? Or did you mean "always false"?

**Comment by patrickjuchli** [comment]
@jeremyml what's the status here?

**Comment by patrickjuchli** [comment]
Addressed in 4.5.0, reopen if necessary.
